Address Safer CPP casting warnings in WebGPUDowncastConvertToBackingContext.cpp

cdumez opened the pull request:
#### 74f36b25384f0d2a8ecd7e0b467a47532103e5c5
<pre>
Address Safer CPP casting warnings in WebGPUDowncastConvertToBackingContext.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=301248">https://bugs.webkit.org/show_bug.cgi?id=301248</a>

Reviewed by Mike Wyrzykowski.

* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPU.h:
(WebCore::WebGPU::GPU::isRemoteGPUProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUAdapter.h:
(WebCore::WebGPU::Adapter::isRemoteAdapterProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUBindGroup.h:
(WebCore::WebGPU::BindGroup::isRemoteBindGroupProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUBindGroupLayout.h:
(WebCore::WebGPU::BindGroupLayout::isRemoteBindGroupLayoutProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUBuffer.h:
(WebCore::WebGPU::Buffer::isRemoteBufferProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCommandBuffer.h:
(WebCore::WebGPU::CommandBuffer::isRemoteCommandBufferProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCommandEncoder.h:
(WebCore::WebGPU::CommandEncoder::isRemoteCommandEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCompositorIntegration.h:
(WebCore::WebGPU::CompositorIntegration::isRemoteCompositorIntegrationProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUComputePassEncoder.h:
(WebCore::WebGPU::ComputePassEncoder::isRemoteComputePassEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUComputePipeline.h:
(WebCore::WebGPU::ComputePipeline::isRemoteComputePipelineProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUExternalTexture.h:
(WebCore::WebGPU::ExternalTexture::isRemoteExternalTextureProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUPipelineLayout.h:
(WebCore::WebGPU::PipelineLayout::isRemotePipelineLayoutProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUPresentationContext.h:
(WebCore::WebGPU::PresentationContext::isRemotePresentationContextProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUQuerySet.h:
(WebCore::WebGPU::QuerySet::isRemoteQuerySetProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUQueue.h:
(WebCore::WebGPU::Queue::isRemoteQueueProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderBundle.h:
(WebCore::WebGPU::RenderBundle::isRemoteRenderBundleProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderBundleEncoder.h:
(WebCore::WebGPU::RenderBundleEncoder::isRemoteRenderBundleEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassEncoder.h:
(WebCore::WebGPU::RenderPassEncoder::isRemoteRenderPassEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPipeline.h:
(WebCore::WebGPU::RenderPipeline::isRemoteRenderPipelineProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUSampler.h:
(WebCore::WebGPU::Sampler::isRemoteSamplerProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUShaderModule.h:
(WebCore::WebGPU::ShaderModule::isRemoteShaderModuleProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUTexture.h:
(WebCore::WebGPU::Texture::isRemoteTextureProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUTextureView.h:
(WebCore::WebGPU::TextureView::isRemoteTextureViewProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUXRBinding.h:
(WebCore::WebGPU::XRBinding::isRemoteXRBindingProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUXRSubImage.h:
(WebCore::WebGPU::XRSubImage::isRemoteXRSubImageProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUXRView.h:
(WebCore::WebGPU::XRView::isRemoteXRViewProxy const):
* Source/WebKit/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteAdapterProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBindGroupLayoutProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBindGroupProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBufferProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCommandBufferProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCommandEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCompositorIntegrationProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePassEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePipelineProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteExternalTextureProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteGPUProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePipelineLayoutProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePresentationContextProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQuerySetProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPipelineProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteSamplerProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteShaderModuleProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteTextureProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteTextureViewProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteXRBindingProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteXRSubImageProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteXRViewProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/WebGPUDowncastConvertToBackingContext.cpp:
(WebKit::WebGPU::DowncastConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::DowncastConvertToBackingContext::convertToRawBacking):

Canonical link: <a href="https://commits.webkit.org/302009@main">https://commits.webkit.org/302009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/63a5a665e16ca12af0acbb9d4635194b0bd4b88b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127573 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47221 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38368 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134650 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/79130 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0795c880-2812-4830-971e-f5291aad8d9b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/129445 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47842 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55749 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/97111 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/65027 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/42aa7874-d702-4191-b18c-b7bcd48606b6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130521 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38262 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114259 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77591 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c9a15c5c-a1b0-4f4c-a26c-462d4e1f2384) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/37076 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/32356 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/78023 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/108128 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32796 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/137134 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/54232 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41795 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105638 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54743 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110616 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105289 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26886 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50803 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/29243 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51818 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/54169 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/60256 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53403 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56860 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/55162 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (approved):
Looks good. The other option would be to use the concrete type and then there is no need for the downcast at all. I filed https://bugs.webkit.org/show_bug.cgi?id=301276 to track implementing that.

cdumez committed:
Address Safer CPP casting warnings in WebGPUDowncastConvertToBackingContext.cpp
https://bugs.webkit.org/show_bug.cgi?id=301248

Reviewed by Mike Wyrzykowski.

* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPU.h:
(WebCore::WebGPU::GPU::isRemoteGPUProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUAdapter.h:
(WebCore::WebGPU::Adapter::isRemoteAdapterProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUBindGroup.h:
(WebCore::WebGPU::BindGroup::isRemoteBindGroupProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUBindGroupLayout.h:
(WebCore::WebGPU::BindGroupLayout::isRemoteBindGroupLayoutProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUBuffer.h:
(WebCore::WebGPU::Buffer::isRemoteBufferProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCommandBuffer.h:
(WebCore::WebGPU::CommandBuffer::isRemoteCommandBufferProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCommandEncoder.h:
(WebCore::WebGPU::CommandEncoder::isRemoteCommandEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUCompositorIntegration.h:
(WebCore::WebGPU::CompositorIntegration::isRemoteCompositorIntegrationProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUComputePassEncoder.h:
(WebCore::WebGPU::ComputePassEncoder::isRemoteComputePassEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUComputePipeline.h:
(WebCore::WebGPU::ComputePipeline::isRemoteComputePipelineProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUExternalTexture.h:
(WebCore::WebGPU::ExternalTexture::isRemoteExternalTextureProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUPipelineLayout.h:
(WebCore::WebGPU::PipelineLayout::isRemotePipelineLayoutProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUPresentationContext.h:
(WebCore::WebGPU::PresentationContext::isRemotePresentationContextProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUQuerySet.h:
(WebCore::WebGPU::QuerySet::isRemoteQuerySetProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUQueue.h:
(WebCore::WebGPU::Queue::isRemoteQueueProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderBundle.h:
(WebCore::WebGPU::RenderBundle::isRemoteRenderBundleProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderBundleEncoder.h:
(WebCore::WebGPU::RenderBundleEncoder::isRemoteRenderBundleEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassEncoder.h:
(WebCore::WebGPU::RenderPassEncoder::isRemoteRenderPassEncoderProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPipeline.h:
(WebCore::WebGPU::RenderPipeline::isRemoteRenderPipelineProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUSampler.h:
(WebCore::WebGPU::Sampler::isRemoteSamplerProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUShaderModule.h:
(WebCore::WebGPU::ShaderModule::isRemoteShaderModuleProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUTexture.h:
(WebCore::WebGPU::Texture::isRemoteTextureProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUTextureView.h:
(WebCore::WebGPU::TextureView::isRemoteTextureViewProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUXRBinding.h:
(WebCore::WebGPU::XRBinding::isRemoteXRBindingProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUXRSubImage.h:
(WebCore::WebGPU::XRSubImage::isRemoteXRSubImageProxy const):
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPUXRView.h:
(WebCore::WebGPU::XRView::isRemoteXRViewProxy const):
* Source/WebKit/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteAdapterProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBindGroupLayoutProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBindGroupProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBufferProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCommandBufferProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCommandEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCompositorIntegrationProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePassEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteComputePipelineProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteExternalTextureProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteGPUProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePipelineLayoutProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePresentationContextProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQuerySetProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPipelineProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteSamplerProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteShaderModuleProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteTextureProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteTextureViewProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteXRBindingProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteXRSubImageProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteXRViewProxy.h:
(isType):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/WebGPUDowncastConvertToBackingContext.cpp:
(WebKit::WebGPU::DowncastConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::DowncastConvertToBackingContext::convertToRawBacking):

Canonical link: https://commits.webkit.org/302009@main
